src/black/files.py

```diff
@@ -103,14 +106,93 @@ def find_pyproject_toml(path_search_start: Tuple[str, ...]) -> Optional[str]:
 
 @mypyc_attr(patchable=True)
 def parse_pyproject_toml(path_config: str) -> Dict[str, Any]:
-    """Parse a pyproject toml file, pulling out relevant parts for Black
+    """Parse a pyproject toml file, pulling out relevant parts for Black.
 
-    If parsing fails, will raise a tomllib.TOMLDecodeError
+    If parsing fails, will raise a tomllib.TOMLDecodeError.
     """
     with open(path_config, "rb") as f:
         pyproject_toml = tomllib.load(f)
-    config = pyproject_toml.get("tool", {}).get("black", {})
-    return {k.replace("--", "").replace("-", "_"): v for k, v in config.items()}
+    config: Dict[str, Any] = pyproject_toml.get("tool", {}).get("black", {})
+    config = {k.replace("--", "").replace("-", "_"): v for k, v in config.items()}
+
+    if "target_version" not in config:
+        inferred_target_version = infer_target_version(pyproject_toml)
+        if inferred_target_version is not None:
+            config["target_version"] = [inferred_target_version.name.lower()]
+
+    return config
+
+
+def infer_target_version(pyproject_toml: Dict[str, Any]) -> Optional[TargetVersion]:
+    """Infer Black's target version from the project metadata in pyproject.toml.
+
+    If target version cannot be inferred, returns None.
+    """
+    requires_python = get_req_python(pyproject_toml)
+    if requires_python is not None:
+        return parse_req_python(requires_python)
+    return None
+
+
+def get_req_python(pyproject_toml: Dict[str, Any]) -> Optional[str]:
+    """Get the required Python version from the project metadata.
+
+    Currently only supports the PyPA standard format:
+    https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#requires-python
+
+    If the field is not present or cannot be parsed, returns None.
+    """
+    project_metadata = pyproject_toml.get("project", {})
+    requires_python: Optional[str] = project_metadata.get("requires-python", None)
+    return requires_python
```

Comment:
```suggestion
    return project_metadata.get("requires-python", None)
```

Reply:
Unfortunately, mypy doesn't like your suggestion:

```
error: Returning Any from function declared to return "Optional[str]"  [no-any-return]
```

That's why I explicitly cast it to an optional string.